Remove CSV file processing function

Harriethw opened the pull request:
<!-- markdownlint-disable-next-line first-line-heading -->
## Description

https://nhsd-jira.digital.nhs.uk/browse/DTOSS-9913

This function was previously introduced to handle CSV files for the pilot. There are no plans to use this in the future and it affects local development setup, so we are removing it.

<!-- Describe your changes in detail. -->

## Context

<!-- Why is this change required? What problem does it solve? -->

## Type of changes

<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->

- [x] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

<!-- Go over all the following points, and put an `x` in all the boxes that apply. -->

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

steventux reviewed (commented):
Looks great and very thorough, adding a test for the migration trigger is :1st_place_medal: 
Only thing I would advise is to remove the `end-to-end` compose profile label here https://github.com/NHSDigital/dtos-communication-management/blob/main/compose.yml#L54 as I don't think we need to spin up an Azurite container as part of the end to end tests now.

Harriethw commented:
> to remove the `end-to-end` compose profile label here https://github.com/NHSDigital/dtos-communication-management/blob/main/compose.yml#L54 as I don't think we need to spin up an Azurite container as part of the end to end tests now.

@steventux thanks for taking a look! I think I've [removed the compose stuff here](https://github.com/NHSDigital/dtos-communication-management/pull/276/commits/8c651f30fd0ed991cbfcefb464036d999eda7f0f), and removed the env variables that weren't needed as well. I also noticed some more stray files that I've removed here if you would be able to double check https://github.com/NHSDigital/dtos-communication-management/pull/276/commits/23757a30c8614a4e4cce0beebddcc79db4950740
